### PR TITLE
cleanup mixerplugin

### PR DESCRIPTION
**Pull request by kyessenov**
Undo the ugly "fix".

Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

**Comment by costinm** [comment]
This break upgrade, which is a requirement. 

/hold

**Comment by kyessenov** [comment]
Note the branch.

**Comment by kyessenov** [comment]
Since my fix in upstream envoy, there is no need to worry about extra fields. This is meant for 1.1+. All the ugly workaround are unnecessary for 1.0 to 1.1 transition.

**Comment by costinm** [comment]
Ok for 1.1 - but it may be better to focus on using the version that is now available and have a proper implementation, I don't want a repeat where 1.0 can't be upgraded to 1.1 because mixer added new attributes or changed config.

Also: everyone is supposed to focus on testing and fixing 1.0, there are plenty of broken things.

**Comment by kyessenov** [comment]
This is due to lack of testing for upgrade. The fix was just one line in
envoy. We just found it out late.

**Comment by rshriram** [comment]
Why can’t that Envoy be pushed to 1.0?

**Comment by kyessenov** [comment]
We did, so transition from 1.0 is smooth. The problem is with existing 0.8
envoy.

**Comment by kyessenov** [comment]
Can we merge this to master? We will not pick up 1.0 changes to mixerfilter since they are not necessary for 1.0+...

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 1]
```diff
@@ -185,22 +184,12 @@ func buildTransport(mesh *meshconfig.MeshConfig, uid attribute) *mccpb.Transport
...
 	}
 
-	res := &mccpb.TransportConfig{
-		CheckCluster:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(policy), port),
-		ReportCluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(telemetry), port),
+	return &mccpb.TransportConfig{
+		CheckCluster:            model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(policy), port),
+		ReportCluster:           model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(telemetry), port),
+		NetworkFailPolicy:       &mccpb.NetworkFailPolicy{Policy: mccpb.FAIL_CLOSE},
+		AttributesForMixerProxy: &mpb.Attributes{Attributes: attributes{"source.uid": uid}},
```
Comment (hklai) [inline pilot/pkg/networking/plugin/mixer/mixer.go]: should there be a test for this?

I suppose os.Getenv("MIXER_NEW_ATTRIBUTES") == "" by default so I suppose there should be a new test to make sure this attribute take effect?

Comment (kyessenov) [reply pilot/pkg/networking/plugin/mixer/mixer.go]: If you look at the history, Costin added this condition without actually using the environment variables anywhere. There is only one match for the use of this variable, and it's here. I'm trying to restore to how things were pre-change.

Comment (hklai) [reply pilot/pkg/networking/plugin/mixer/mixer.go]: That I understand. I am just wondering why adding and removing that condition does not involve fixing a test? It seems to mean we are missing a test for it.

**Comment by kyessenov** [comment]
There is no automated test unfortunately.

File: pilot/pkg/networking/plugin/mixer/mixer.go (hunk 1 above)
Comment (kyessenov) [reply pilot/pkg/networking/plugin/mixer/mixer.go]: There is no tests for this feature. I think the best we have is mixer tests but they are not sufficiently detailed to validate that _all_ of telemetry is present.

**Comment by kyessenov** [comment]
@hklai yes, it can. I'm just worried this PR is going to be aggressively closed, and since there is no test, we will forget to clean it up again.

**Comment by hklai** [comment]
Hopefully we can open up next week.

And the state bot will nag this before it closes.

**Comment by kyessenov** [comment]
I give up. Pushed another change to release branch.

**Pull request closed without merging**